cloud/pkg/synccontroller/objectsync.go

```diff
@@ -34,46 +36,59 @@ func (sctl *SyncController) manageObject(sync *v1alpha1.ObjectSync) {
 	ret, err := sctl.kubeclient.Resource(gvr).Namespace(sync.Namespace).Get(context.TODO(), sync.Spec.ObjectName, metav1.GetOptions{})
 	if err != nil || ret == nil {
 		klog.Errorf("failed to get obj(gvr:%v,namespace:%v,name:%v), %v", gvr, sync.Namespace, sync.Spec.ObjectName, err)
+		sendFailEvents(sync, err)
 		return
 	}
 
-	nodeName := getNodeName(sync.Name)
-
 	object, err = meta.Accessor(ret)
 	if err != nil {
+		klog.Errorf("failed to get obj with meta (gvr:%v,namespace:%v,name:%v), %v", gvr, sync.Namespace, sync.Spec.ObjectName, err)
+		sendFailEvents(sync, err)
 		return
 	}
-
 	syncObjUID := getObjectUID(sync.Name)
 	if syncObjUID != string(object.GetUID()) {
 		err = apierrors.NewNotFound(schema.GroupResource{
 			Group:    "",
 			Resource: sync.Spec.ObjectKind,
 		}, sync.Spec.ObjectName)
+		sendFailEvents(sync, err)
```

Comment:
just question：why sendFailEvent here to send delete msg to edge-node? metaserver will delete from meta db according to msg.key not obj.uid.

Reply:
This is managed by `metamanager`